js/pages/tola_management_pages/user/components/edit_user_programs.js

```diff
@@ -25,15 +25,31 @@ const create_country_objects = (countries, store) => Object.entries(countries)
                                                         }
                                                     }),{})
 
-const create_program_objects = (programs, store) => Object.entries(programs)
-                                                           .reduce((programs, [id, program]) => ({
-                                                               ...programs,
-                                                               [id]: {
-                                                                   ...program,
-                                                                   type: 'program',
-                                                                   options: [{label: NO_ACCESS, value: 'none'}, ...store.program_role_choices],
-                                                               }
-                                                           }),{})
+// generates the programs objects for the virtualized table, with appropriate menu options
+const create_program_objects = (programs, store) => {
+    // first get a list of programs to which the user has country-level access to:
+    const programsWithCountryAccess = Object.entries(store.editing_target_data.access.countries)
+            // only count objects with 'user' or 'basic_admin' access (should be only options, this is future-proofing)
+            .filter(([countryId, country]) => country.role == 'basic_admin' || country.role == 'user')
+            // filter out any that aren't in the country store for whatever reason (bug catcher)
+            .filter(([countryId, country]) => store.countries[countryId])
+            // just get the programs list for each country:
+            .map(([countryId, country]) => store.countries[countryId].programs)
+            .flat();
+    return Object.entries(programs)
+        .reduce((programs, [id, program]) => ({
+            ...programs,
+            [id]: {
+                ...program,
+                type: 'program',
+                options: (programsWithCountryAccess.includes(parseInt(id)) ?
+                            // if given country level access, do not include "no access" choice:
+                            [...store.program_role_choices] :
+                            // otherwise "no access" is a choice:
+                            [{label: NO_ACCESS, value: 'none'}, ...store.program_role_choices]),
+            }
+        }),{});
+}
```

Comment:
All of this commenting is super helpful!

Reply:
this was a really frustrating batch of tickets to work through because I had to relearn the whole weird ad hoc data structure of countries/programs each time, so I'm trying to leave us some breadcrumbs to follow next time!